.github/workflows/build.yml

```diff
@@ -1,70 +1,93 @@
 name: "Build"
 on:
-    push:
-        branches:
-            - "4.x"
-        tags:
-            - v4*
-    pull_request:
-        branches:
-            - "4.x"
-        types: [opened, reopened, synchronize]
-jobs:
-    lint:
-        runs-on: ubuntu-latest
-        env:
-            TEST: "build_and_lint"
-            GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        steps:
-            - uses: actions/setup-node@v1
-              with:
-                  node-version: 14
-
-            - uses: actions/checkout@v2
-
-            - run: yarn
-
-            - run: yarn lint
-    unit:
-        runs-on: ubuntu-latest
-        strategy:
-            matrix:
-                node: [ 14, 16 ]
-        steps:
-            - uses: actions/setup-node@v1
-              with:
-                  node-version: ${{ matrix.node }}
-
-            - uses: actions/checkout@v2
+  push:
+    branches:
+      - "4.x"
+    tags:
+      - v4*
+  pull_request:
+    branches:
+      - "4.x"
+    types: [opened, reopened, synchronize]
 
-            - run: yarn
-
-            - run: yarn test:unit
-
-    e2e:
-        runs-on: ubuntu-latest
-        strategy:
-            fail-fast: false
-            matrix:
-                node: [ 14, 16 ]
-                testCmd: [ 'e2e_ganache', 'e2e_geth', 'e2e_browsers' ]
-        env:
-            TEST_CMD: ${{ matrix.testCmd }}
-        steps:
-            -   uses: actions/setup-node@v1
-                with:
-                    node-version: ${{ matrix.node }}
-
-            -   uses: actions/checkout@v2
-
-            -   run: yarn
+jobs:
+  build: 
+    name: Build Packages
+    runs-on: ubuntu-latest
+    steps: 
+      - uses: actions/checkout@v3    
+      - uses: actions/setup-node@v3
+        with: 
+          node-version: 14
+          cache: yarn
+      - run: yarn
 
-            -   if: (env.TEST_CMD == 'e2e_ganache')
-                run: yarn ganache-background && yarn test:integration
+  lint:
+    name: Lint
+    needs: build
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Curious as to why this is needed for the linter?

Reply:
I personally dont' know. I believe it was copied over from previous configurations.

Reply:
I think we should try removing it, probably necessary for something in 1.x, but I don't think it's needed for 4.x